backend/users/admin.py

```diff
@@ -35,7 +56,46 @@ class CustomUserAdmin(UserAdmin):
     ordering = ("name", "email")
 
 
+class BulkImportAdmin(admin.ModelAdmin):
+    change_list_template = "admin/csv_change_list.html"
+
+    def get_urls(self):
+        urls = super().get_urls()
+        my_urls = [
+            path("import-csv/", self.import_csv),
+        ]
+        return my_urls + urls
+
+    def import_csv(self, request):
+        if request.method == "POST":
+            try:
+                csv_file = TextIOWrapper(request.FILES['csv_file'].file, encoding=request.encoding)
+                dialect = csv.Sniffer().sniff(csv_file.read(1024))
+                csv_file.seek(0)
+                reader = csv.DictReader(csv_file, dialect=dialect)
+            except Exception as err:
+                self.message_user(request, "Error: {}".format(err))
+                return redirect("..")
+            try:
+                for row in reader:
+                    if ('/student/' in request.path):
```

Comment:
It would be better to capture the url path, and send it as keyword argument to the view.

Reply:
Could you elaborate a little, I didn't get you.

Reply:
Ignore my earlier point.

However, this `in` lookup is very costly, as it is run for every single row in the csv, but it remains constant. Better to swap the for and if statements. (You'll have to duplicate the for statement)